denote.el

```diff
@@ -825,35 +851,34 @@ leading and trailing hyphen."
   (replace-regexp-in-string "\\." "" str))
 
 (defun denote--trim-right-token-characters (str component)
-  "Remove =, - and _ from the end of STR.
+  "Remove =, -, _ and @ from the end of STR.
 The removal is done only if necessary according to COMPONENT."
   (if (eq component 'title)
-      (string-trim-right str "[=_]+")
-    (string-trim-right str "[=_-]+")))
+      (string-trim-right str "[=@_]+")
+    (string-trim-right str "[=@_-]+")))
 
 (defun denote--replace-consecutive-token-characters (str component)
   "Replace consecutive characters with a single one in STR.
-Hyphens, underscores and equal signs are replaced with a single
-one in str, if necessary according to COMPONENT."
-  ;; -- are allowed in titles
-  (if (eq component 'title)
-      (replace-regexp-in-string
-       "_\\{2,\\}" "_"
-       (replace-regexp-in-string
-        "=\\{2,\\}" "=" str))
-    (replace-regexp-in-string
-     "-\\{2,\\}" "-"
-     (replace-regexp-in-string
-      "_\\{2,\\}" "_"
+Hyphens, underscores, equal signs and at signs are replaced with
+a single one in str, if necessary according to COMPONENT."
+  (let ((str (replace-regexp-in-string
+              "_\\{2,\\}" "_"
+              (replace-regexp-in-string
+               "=\\{2,\\}" "="
+               (replace-regexp-in-string
+                "@\\{2,\\}" "@" str)))))
+    ;; -- are allowed in titles
```

Comment:
Are these allowed unconditionally? I just tried it and the consecutive dashes were not in the file name (our current behaviour). Anyway, the consecutive dashes should be opt-in.